File: skills/astra-toolkit/clients/typescript/examples/admin-find-reranking-providers.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

const client = new DataAPIClient(process.env.ASTRA_DB_APPLICATION_TOKEN!);

const admin = client.admin();

const databaseAdmin = admin.dbAdmin(process.env.ASTRA_DB_API_ENDPOINT!);

(async function () {
  const providers = await databaseAdmin.findRerankingProviders();

  console.log(JSON.stringify(providers));
})();

File: skills/astra-toolkit/clients/python/examples/admin-find-reranking-providers.py
import os
from astrapy import DataAPIClient

client = DataAPIClient(os.environ["ASTRA_DB_APPLICATION_TOKEN"])

admin = client.get_admin()

database_admin = admin.get_database_admin(os.environ["ASTRA_DB_API_ENDPOINT"])

providers = database_admin.find_reranking_providers()

# Use the raw dict
print(providers.raw_info)

# Or work with the resulting object:

print(providers.reranking_providers.keys())

print(providers.reranking_providers["nvidia"])

print(providers.reranking_providers["nvidia"].models[0])
